history.go

```diff
@@ -0,0 +1,75 @@
+package merkletree
+
+import (
+	"errors"
+
+	"github.com/coniks-sys/libmerkleprefixtree-go/crypto"
+)
+
+var (
+	ErrHistoryExisted = errors.New("[merkletree] trying to override existing history")
+	ErrBadEpoch       = errors.New("[merkletree] next epoch's STR has bad epoch ")
+)
+
+type History struct {
+	tree          *MerkleTree
+	currentSTR    *SignedTreeRoot
+	epochInterval int64
+}
```

Comment:
```epoch``` and ```epochInterval``` are different things. ```epoch``` is the actual time at which the next history is updated, whereas ```epochInterval``` is the frequency at which the history is updated. Might be worth renaming the latter to ```updateFrequency``` or something like that.